PANDORA/Pandora/Modelling_functions.py

```
@@ -122,93 +122,101 @@ def predict_anchors_netMHCIIpan(peptide, allele_type, verbose=True):
-        return [3, 6, 8, 11]
+        except ValueError:
+            print('Could not predict binding core using netMHCIIpan. Will use the most common anchor positions instead')
+            return [3, 6, 8, 11]
 
-    # Remove output file
-    os.system('rm %s %s' % (infile, outfile))
+        # Remove output file
+        os.system('rm %s %s' % (infile, outfile))
 
-    offset, core, core_reliability = max_scores[0]
-    # Use the canonical spacing for 9-mer binding cores to predict the anchor positions
-    predicted_anchors = [offset + 1, offset + 4, offset + 6, offset + 9]
-    # Make sure the prediction is not longer than the peptide just in case
-    predicted_anchors = [i for i in predicted_anchors if i <= len(peptide)]
+        offset, core, core_reliability = max_scores[0]
+        # Use the canonical spacing for 9-mer binding cores to predict the anchor positions
+        predicted_anchors = [offset + 1, offset + 4, offset + 6, offset + 9]
+        # Make sure the prediction is not longer than the peptide just in case
+        predicted_anchors = [i for i in predicted_anchors if i <= len(peptide)]
 
-    if verbose:
-        print('\tPredicted the binding core using netMHCIIpan (4.0):\n')
-        print('\toffset:\t%s\n\tcore:\t%s\n\tprob:\t%s\n' % (offset, core, core_reliability))
-        print('\tPredicted peptide anchor residues (assuming canonical spacing): %s' % predicted_anchors)
+        if verbose:
+            print('\tPredicted the binding core using netMHCIIpan (4.0):\n')
+            print('\toffset:\t%s\n\tcore:\t%s\n\tprob:\t%s\n' % (offset, core, core_reliability))
+            print('\tPredicted peptide anchor residues (assuming canonical spacing): %s' % predicted_anchors)
 
-    return predicted_anchors
+        return predicted_anchors
+    
+    else:
+        print("Need netMHCIIpan to predict anchor positions. Please download and install netMHCIIpan.\n\n"
```

@Smaassen2 here (both for class I and II) the code should raise an exception and close the run after printing this message, it should not just print a message and send the code on, otherwise it will crash later with a much more unclear error.